.github/workflows/actionlint.yml

```diff
@@ -0,0 +1,52 @@
+name: actionlint
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - '.github/workflows/*.yml'
+      - '.github/workflows/*.yaml'
+  pull_request:
+    paths:
+      - '.github/workflows/*.yml'
+      - '.github/workflows/*.yaml'
+  merge_group:
+
+env:
+  HOMEBREW_DEVELOPER: 1
+
+concurrency:
+  group: "actionlint-${{ github.ref }}"
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+jobs:
+  workflow_syntax:
+    if: github.repository == 'Homebrew/brew'
+    runs-on: ubuntu-22.04
+    container:
+      image: ghcr.io/homebrew/ubuntu22.04:master
+    defaults:
+      run:
+        shell: /bin/bash -e {0}
```

Comment:
The default is `bash -e {0}`. Are we avoiding another bash somewhere?

Reply:
IIUC correctly in containers the default is `sh`. Or an `sh` masquerading as `bash`.

Reply:
Docs appear to suggest `sh` is only used when `bash` is not found: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell

Reply:
Hmm, weird. I know I've had trouble with `[[`, `]]` in Homebrew/core workflows when I don't set `shell: /bin/bash`.

Reply:
Now getting
```
shellcheck reported issue in this script: SC2115:warning:4:27: Use "${var:?}" to ensure this never expands to /* 
```
After dropping this.

Reply:
It's possible it might just not run shellcheck with custom shells, which could make sense.

Reply:
I get
```
/__w/_temp/b13f8fe9-518c-4fd3-af62-a31519cf8e56.sh: 5: shopt: not found
```
when running without setting the shell to `/bin/bash` in Homebrew/core. Must have something to do with running in a container, since the shell there is reported as `sh -e {0}`.

https://github.com/Homebrew/homebrew-core/actions/runs/4678530113/jobs/8287369304?pr=128075#step:5:34